File: CPlus/__ltpService/LTPService.cpp
#include "LTPService.h"

namespace HIT_IR_LTP{
	LTPService::~LTPService(void)
	{
	}

	bool LTPService::Analyze(const std::string& option, const std::string& analyzeString, LTML& ltml_out)
	{
		int flag = -1;
		ls.SetAnalysisOptions(option);
		ls.SetXmlOption(false);
		std::string resultStr;
		if(analyzeString.size() == 0)
		{
			std::cerr<<"Input analyzeString is null!"<<std::endl;
		}
		else
		{
			//		printf("analyzeString: \n%s\n", analyzeString.c_str());
			flag = ls.Analyze(analyzeString, resultStr);
			//		printf("result: \n%s\n", resultStr.c_str());
			ltml_out.ClearDOM();
			ltml_out.SetEncoding(ls.GetEncoding());
			ltml_out.LoadLtml(resultStr);
			ltml_out.SetOver();
		}
		if (flag==0)
		{
			return true;
		}
		return false;
	}

	bool LTPService::Analyze(const std::string& option, const LTML& ltml_in, LTML& ltml_out)
	{
		int flag = -1;
		ls.SetAnalysisOptions(option);
		ls.SetXmlOption(true);
		std::string resultStr;
		flag = ls.Analyze(ltml_in.GetXMLStr(), resultStr);
		ltml_out.ClearDOM();
		ltml_out.SetEncoding(ls.GetEncoding());
		ltml_out.LoadLtml(resultStr);
		ltml_out.SetOver();
		if (flag==0)
		{
			return true;
		}
		return false;
	}
}

File: C#/LTPService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ltp_service
{
    public class LTPService
    {
        #region Variables

        private CirService m_cirServ = null;

        #endregion

        #region Constructions

        public LTPService(String strAuthor)
        {
            m_cirServ = new CirService(strAuthor);
            Authorize(strAuthor);
        }

        #endregion

        #region Public Methods


        public bool SetEncoding(String encodeType)
        {
            try
            {
                m_cirServ.setEncoding(encodeType);
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
                return false;
            }

            return true;          
        }

        public LTML Analyze(String strOption, String strAnalyzeString)
        {
            if (strOption == null || strAnalyzeString == null)
            {
                throw new Exception(String.Format("{0}: LTPService.Analyze(strOption, strAnalyzeString)", LtpServiceError.ParameterNullError));
            }
            m_cirServ.SetAnalysisOptions(strOption);
		    m_cirServ.SetXmlOption(false);
            LTML ltml_out = new LTML();
            String resultStr = m_cirServ.Analyze(strAnalyzeString);
            ltml_out.LoadFromXmlStr(resultStr);
            return ltml_out;
        }

        public LTML Analyze(String strOption, LTML ltmlIn)
        {
            if (strOption == null || ltmlIn == null)
            {
                throw new Exception(String.Format("{0}: LTPService.Analyze(strOption, ltmlIn)", LtpServiceError.ParameterNullError));
            }
            m_cirServ.SetAnalysisOptions(strOption);
            m_cirServ.SetXmlOption(true);
            String resultStr = m_cirServ.Analyze(ltmlIn.GetXMLStr());
            LTML ltml_out = new LTML();
            ltml_out.LoadFromXmlStr(resultStr);
            return ltml_out;
        }

        #endregion

        #region Private Methods

        private int Authorize(String strAuthor)
        {
            try
            {
                m_cirServ.Authorize(strAuthor);
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
                return 0;
            }

            return 1;
        }

        #endregion
    }
}
